.github/workflows/update-actions.yml

```
@@ -0,0 +1,41 @@
+          echo "Checking for actions using @master or @main tags..."
+          # Exclude snyk/actions/docker as it only has @master available
+          if grep -rh "uses:" .github/workflows/*.yml | grep -E "@master|@main" | grep -v "snyk/actions/docker"; then
+            echo "⚠️ Warning: Found actions using @master or @main tags"
+            echo "Consider pinning to specific versions for better security and reproducibility"
+            exit 1
+          else
+            echo "✅ All actions are using versioned tags (or approved exceptions)"
+          fi
+
+      - name: List all action versions
+        run: |
```

actionlint (1.7.12)

[error] shellcheck reported issue in this script: SC2086:info:1:37: Double quote to prevent globbing and word splitting

(shellcheck)